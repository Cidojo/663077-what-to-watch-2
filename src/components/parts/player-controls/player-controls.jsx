import * as React from 'react';
import PropTypes from 'prop-types';

import {Button} from './../button/button.jsx';
import {PlayerTime} from './../player-time/player-time.jsx';
import withTimeline from './../../../hocs/with-timeline/with-timeline.jsx';

import {Icon} from './../../../constants/constants.js';

const PlayButtonText = {
  PLAY: `Play`,
  PAUSE: `Pause`
};

const PlayerControls = (props) => {
  const {
    isPlaying,
    onFullscreenButtonClick,
    onPlayToggle,
    secondsPlayed,
    duration,
    movieTitle
  } = props;

  return (
    <div className="player__controls">
      <div className="player__controls-row">
        <PlayerTime
          duration={duration}
          currentTime={secondsPlayed}
        />
      </div>

      <div className="player__controls-row">
        <Button
          className="player__play"
          icon={isPlaying ? Icon.PAUSE : Icon.PLAY}
          onClick={onPlayToggle}
        >
          {isPlaying ? PlayButtonText.PAUSE : PlayButtonText.PLAY}
        </Button>

        <div className="player__name">{duration ? movieTitle : `Transpotting`}</div>

        <Button
          className="player__full-screen"
          icon={Icon.FULL_SCREEN}
          onClick={onFullscreenButtonClick}
        >
          Full screen
        </Button>
      </div>
    </div>
  );
};

PlayerControls.propTypes = {
  isPlaying: PropTypes.oneOf([true, false, null]),
  duration: PropTypes.number.isRequired,
  secondsPlayed: PropTypes.number.isRequired,
  movieTitle: PropTypes.string.isRequired,
  onPlayToggle: PropTypes.func.isRequired,
  onFullscreenButtonClick: PropTypes.func.isRequired
};

PlayerControls.defaultProps = {
  isPlaying: false,
  secondsPlayed: 0,
  duration: 0,
  movieTitle: ``,
  onPlayToggle: () => {},
  onFullscreenButtonClick: () => {}
};

export {PlayerControls};
export default withTimeline(PlayerControls);
