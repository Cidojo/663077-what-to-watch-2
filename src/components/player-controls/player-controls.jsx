import * as React from 'react';
import PropTypes from 'prop-types';
import {Button} from './../button/button.jsx';
import {PlayerTime} from './../player-time/player-time.jsx';

const PlayerControls = (props) => {
  const {
    isPlaying,
    onFullscreenButtonClick,
    onPlayButtonClick,
    currentTime,
    duration,
    movieTitle
  } = props;


  return (
    <div className="player__controls">
      <div className="player__controls-row">
        <PlayerTime
          duration={duration}
          currentTime={currentTime}
        />
      </div>

      <div className="player__controls-row">
        <Button
          className="player__play"
          icon={isPlaying ? `PAUSE` : `PLAY`}
          onClick={onPlayButtonClick}
        >
          {isPlaying ? `Pause` : `Play`}
        </Button>

        <div className="player__name">{duration ? movieTitle : `Transpotting`}</div>

        <Button
          className="player__full-screen"
          icon="FULL_SCREEN"
          onClick={onFullscreenButtonClick}
        >
          Full screen
        </Button>
      </div>
    </div>
  );
};

PlayerControls.propTypes = {
  isPlaying: PropTypes.oneOf([true, false, null]),
  duration: PropTypes.number.isRequired,
  currentTime: PropTypes.number.isRequired,
  movieTitle: PropTypes.string.isRequired,
  onPlayButtonClick: PropTypes.func.isRequired,
  onFullscreenButtonClick: PropTypes.func.isRequired
};

PlayerControls.defaultProps = {
  isPlaying: false,
  currentTime: 0,
  duration: 0,
  movieTitle: ``,
  onPlayButtonClick: () => {},
  onFullscreenButtonClick: () => {}
};

export {PlayerControls};
