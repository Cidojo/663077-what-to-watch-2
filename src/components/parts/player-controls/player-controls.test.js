import * as React from 'react';
import {PlayerControls} from './player-controls.jsx';
import * as renderer from 'react-test-renderer';

const props = {
  isPlaying: false,
  secondsPlayed: 0,
  duration: 0,
  movieTitle: ``,
  onPlayToggle: () => {},
  onFullscreenButtonClick: () => {}
};

it(`should render PlayerControls component without errors`, () => {
  const tree = renderer.create(<PlayerControls {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
