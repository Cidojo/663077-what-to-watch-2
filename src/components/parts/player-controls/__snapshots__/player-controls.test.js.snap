// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlayerControls component without errors 1`] = `
<div
  className="player__controls"
>
  <div
    className="player__controls-row"
  >
    <div
      className="player__time"
    >
      <progress
        className="player__progress"
        max="100"
        value={0}
      />
      <div
        className="player__toggler"
        style={
          Object {
            "left": "0%",
          }
        }
      >
        Toggler
      </div>
    </div>
    <div
      className="player__time-value"
    >
      00:00:00
    </div>
  </div>
  <div
    className="player__controls-row"
  >
    <button
      className="player__play"
      onClick={[Function]}
      type="button"
    >
      <svg
        height={19}
        viewBox="0 0 19 19"
        width={19}
      >
        <use
          xlinkHref="#play-s"
        />
      </svg>
      <span>
        Play
      </span>
    </button>
    <div
      className="player__name"
    >
      Transpotting
    </div>
    <button
      className="player__full-screen"
      onClick={[Function]}
      type="button"
    >
      <svg
        height={21}
        viewBox="0 0 21 21"
        width={21}
      >
        <use
          xlinkHref="#full-screen"
        />
      </svg>
      <span>
        Full screen
      </span>
    </button>
  </div>
</div>
`;
